### PR TITLE
new CI tutorial -- publish allure report -- initial commit DOC-2491

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: tutorials/build-code/ci-publish-allure-report.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+---
+sidebar_position: 120
+description: Create a CI pipeline that runs a maven test and generates an allure report that you can view in the Harness UI.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:3]:
```suggestion
description: Create a CI pipeline that runs a Maven test and generates an Allure Report that you can view in the Harness UI.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+keywords: [Hosted Build, Continuous Integration, Hosted, CI Tutorial, maven, Allure]
+---
+
+# Publish an allure report to the Artifacts tab
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:7]:
```suggestion
# Publish an Allure Report to the Artifacts tab 
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
+---
+sidebar_position: 120
+description: Create a CI pipeline that runs a maven test and generates an allure report that you can view in the Harness UI.
+keywords: [Hosted Build, Continuous Integration, Hosted, CI Tutorial, maven, Allure]
+---
+
+# Publish an allure report to the Artifacts tab 
+
+This tutorial provides an example pipeline that builds a Java maven application and generates an allure report that you can view in the Harness UI.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:9]:
```suggestion
This tutorial provides an example pipeline that builds a Java Maven application and generates an Allure Report that you can view in the Harness UI. 
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+keywords: [Hosted Build, Continuous Integration, Hosted, CI Tutorial, maven, Allure]
+---
+
+# Publish an allure report to the Artifacts tab 
+
+This tutorial provides an example pipeline that builds a Java maven application and generates an allure report that you can view in the Harness UI. 
+
+:::note
+
+Artifact tab publishing is supported in Kubernetes infrastructure only.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:13]:
```suggestion
Artifact tab publishing is supported in the Kubernetes infrastructure only.
```

File: tutorials/build-code/ci-tutorial-go-containers.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
-sidebar_position: 5
-description: This build automation guide walks you through building a Go application and publish the container image of the application using a CI Pipeline
+sidebar_position: 30
+description: This build automation guide describes building a Go container in a a CI Pipeline.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-tutorial-go-containers.md:3]:
```suggestion
description: This build automation guide describes building a Go container in a CI pipeline.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+:::
+
+### Pipeline workflow 
+
+This pipeline has five steps that do the following: 
+
+1. Runs the Maven tests present in the project. 
+
+2. Generates the allure report using the `allure-results` generated as part of step 1. The Allure tool is present in the `solutis/allure:2.9.0` docker image used as part of the step.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:23]:
```suggestion
2. Generates the Allure Report using the `allure-results` generated as part of step 1. The Allure tool is present in the `solutis/allure:2.9.0` Docker image used as part of the step.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+3. Combines the allure report to a single HTML file.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:25]:
```suggestion
3. Combines the Allure Report into a single HTML file.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+   
+   Viewing an allure report in browser requires running a web server via `allure open` command. But this won't allow viewing the report after the CI pipeline is complete.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:27]:
```suggestion
   Viewing an Allure Report in a browser requires running a web server via the `allure open` command. But this won't allow viewing the report after the CI pipeline is complete.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+   We will use [allure-combine](https://pypi.org/project/allure-combine/) tool to convert the allure report into a single HTML file. Running `allure-combine .` inside `allure-report` will generate `complete.html` file. This step uses `shubham149/allure-combine:latest` docker image with `allure-combine` tool present in it.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:29]:
```suggestion
   Use the [allure-combine](https://pypi.org/project/allure-combine/) tool to convert the Allure Report into a single HTML file. Running `allure-combine .` inside `allure-report` generates the `complete.html` file. This step uses the  `shubham149/allure-combine:latest` Docker image with the `allure-combine` tool present in it.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+4. Publishes the report to a  Google Cloud Storage (GCS) bucket.
+ 
+5. Uses static web hosting in GCS to display the allure report in the **Artifacts** tab.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:33]:
```suggestion
5. Uses static web hosting in GCS to display the Allure Report on the **Artifacts** tab. 
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+   
+   Viewing an allure report in browser requires running a web server via `allure open` command. But this won't allow viewing the report after the CI pipeline is complete.
+
+   We will use [allure-combine](https://pypi.org/project/allure-combine/) tool to convert the allure report into a single HTML file. Running `allure-combine .` inside `allure-report` will generate `complete.html` file. This step uses `shubham149/allure-combine:latest` docker image with `allure-combine` tool present in it.
+
+4. Publishes the report to a  Google Cloud Storage (GCS) bucket.
+ 
+5. Uses static web hosting in GCS to display the allure report in the **Artifacts** tab. 
+   
+   `plugins/artifact-metadata-publisher` plugin adds the input urls to artifact tab in Harness UI.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:35]:
```suggestion
   The `plugins/artifact-metadata-publisher` plugin adds the input URLs to the **Artifacts** tab in the Harness UI.
```

File: tutorials/build-code/ci-publish-allure-report.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+
+   - [GitHub Connector](/docs/platform/7_Connectors/connect-to-code-repo.md)
+   - [GCP Connector](/docs/platform/7_Connectors/connect-to-google-cloud-platform-gcp.md) 
+      Use the file secret you just created for the GCP credentials.
+   - [Docker Hub Connector](/docs/platform/7_Connectors/ref-cloud-providers/docker-registry-connector-settings-reference.md)  
+      You can choose to download Harness images from the [Harness Image Registry](docs/platform/7_Connectors/connect-to-harness-container-image-registry-using-docker-connector.md) instead of Docker Hub.
+
+3.  Create a [publicly available bucket in GCS](https://cloud.google.com/storage/docs/access-control/making-data-public#objects) to store the report. 
+
+4.  Update the report URL in steps 4 and 5 with your GCS bucket name ($GCS_BUCKET_NAME).
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-publish-allure-report.md:63]: "in steps 4 and 5" - steps 4 and 5 where? In the Pipeline workflow section? It's not clear. Please revise for clarity.

**Review by KimberlyFields — DISMISSED** [review]
I made a few suggestions and comments, but I'm approving in anticipation of those items being addressed.

**Comment by douglas-j-bothwell** [comment]
FYI @KimberlyFields I've implemented your review comments and am merging this via bypassing branch protections.

**Comment by shubham149** [comment]
@douglas-j-bothwell Connectoref cannot decode environment variables

**Comment by shubham149** [comment]
Same for GCS bucket name. Use stage variables instead

**Pull request merged**